apps/remix-ide/src/app/tabs/compile-tab.js

```diff
@@ -144,17 +144,12 @@ class CompileTab extends CompilerApiMixin(ViewPlugin) { // implements ICompilerA
     this.queryParams.update(params)
   }
 
-  getAppParameter (name) {
-    // first look in the URL params then in the local storage
-    const params = this.queryParams.get()
-    const param = params[name] ? params[name] : this.config.get(name)
-    if (param === 'true') return true
-    if (param === 'false') return false
-    return param
+  async getAppParameter (name) {
```

Comment:
are you certain that all the occurrences of `getAppParameter` use `await` now?

Reply:
Can you find a place where it doesn't use it?

Reply:
I think this is fine.
btw maybe the`getAppParameter` from  `compiler.ts` needs to use the new config plugin then, or is this not part of this PR and can be done afterwards ?

Reply:
I saw that too, it confused me. The other one uses localstorage directly?

Reply:
yes, the impl of the one in the plugin, is too use url param if any and fallback to the localstorage.
I think it's fine if the version from `compiler.ts` uses that version

Reply:
That's better than I can change the api ts definition too.